### ISSUE TITLE
Emergency repair, Xiaomi 8 mobile phone After the screen is off, the game restarts immediately, and the game data is lost immediately.

### ISSUE THREAD

**Issue by chongbo2013** [libgdx/libgdx#5312]
After the screen was off, Xiaomi 8 immediately executed the following code：
android.os.Process.killProcess(android.os.Process.myPid());
Cause application restart

public void pause () {
		synchronized (synch) {
			if (!running) return;
			running = false;
			pause = true;
			while (pause) {
				try {
					// TODO: fix deadlock race condition with quick resume/pause.
					// Temporary workaround:
					// Android ANR time is 5 seconds, so wait up to 4 seconds before assuming
					// deadlock and killing process. This can easily be triggered by opening the
					// Recent Apps list and then double-tapping the Recent Apps button with
					// ~500ms between taps.
					synch.wait(3500);
					if (pause) {
						// pause will never go false if onDrawFrame is never called by the GLThread
						// when entering this method, we MUST enforce continuous rendering
						Gdx.app.error(LOG_TAG, "waiting for pause synchronization took too long; assuming deadlock and killing");
						android.os.Process.killProcess(android.os.Process.myPid());
					}
				} catch (InterruptedException ignored) {
					Gdx.app.log(LOG_TAG, "waiting for pause synchronization failed!");
				}
			}
		}
	}

**Comment by Ameb** [comment]
On a Xiaomi Mix 2S, with similar hw to Xiaomi Mi 8 i am having a similar crash when i lock the screen while the application is on foreground. 

This might be related to [#4626](https://github.com/libgdx/libgdx/issues/4626#issuecomment-405010358)

The crash only occurs when the application is in foreground. If you switch apps and then lock the screen when the phone is awaken the application can be switched back without problems.

Related log:
```
07-25 17:27:22.219 12770 12770 E AndroidGraphics: waiting for pause synchronization took too long; assuming deadlock and killing
07-25 17:27:22.231 12770 12770 I Process : Sending signal. PID: 12770 SIG: 9
```

Sadly i'm just an user of an application which uses libgdx ([mindustry](https://github.com/Anuken/Mindustry)) and can't provide much information.
```
MIUI 9.6 | Stable
Android version 8.0.0
```

**Comment by Ameb** [comment]
The workaround discussed [here](https://github.com/libgdx/libgdx/issues/3861#issuecomment-185684468) (#3861) works.

**Comment by obigu** [comment]
I believe this PR may solve this issue https://github.com/libgdx/libgdx/pull/4326. It'd be great if anynbody who can reproduce the issue could test it.

**Comment by tpnet** [comment]
@Ameb   [this](https://github.com/libgdx/libgdx/issues/3861#issuecomment-185684468) #3861  ， found this exception:

```
2019-02-28 20:10:40.553 13289-13727/com.lxjy.kkk.dyw W/System.err: java.lang.IllegalStateException: PolygonSpriteBatch.begin must be called before end.
2019-02-28 20:10:40.554 13289-13727/com.lxjy.kkk.dyw W/System.err:     at com.badlogic.gdx.graphics.g2d.PolygonSpriteBatch.end(PolygonSpriteBatch.java:168)
2019-02-28 20:10:40.554 13289-13727/com.lxjy.kkk.dyw W/System.err:     at com.badlogic.gdx.scenes.scene2d.Stage.draw(Stage.java:129)
2019-02-28 20:10:40.554 13289-13727/com.lxjy.kkk.dyw W/System.err:     at com.jimu52.fiction.read.mvp.ui.layout.ReadGameViewNew.render(ReadGameViewNew.kt:163)
2019-02-28 20:10:40.554 13289-13727/com.lxjy.kkk.dyw W/System.err:     at com.badlogic.gdx.backends.android.AndroidGraphics.onDrawFrame(AndroidGraphics.java:495)
2019-02-28 20:10:40.554 13289-13727/com.lxjy.kkk.dyw W/System.err:     at com.jimu52.fiction.read.app.PatchedAndroidFragmentApplication$forcePause$1.run(PatchedAndroidFragmentApplication.kt:21)
2019-02-28 20:10:40.554 13289-13727/com.lxjy.kkk.dyw W/System.err:     at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:458)
2019-02-28 20:10:40.554 13289-13727/com.lxjy.kkk.dyw W/System.err:     at java.util.concurrent.FutureTask.run(FutureTask.java:266)
2019-02-28 20:10:40.554 13289-13727/com.lxjy.kkk.dyw W/System.err:     at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
2019-02-28 20:10:40.554 13289-13727/com.lxjy.kkk.dyw W/System.err:     at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
2019-02-28 20:10:40.554 13289-13727/com.lxjy.kkk.dyw W/System.err:     at java.lang.Thread.run(Thread.java:764)
```

**Comment by cypherdare** [comment]
The issue on my Google Pixel was resolved by 1.9.11.

This was most likely fixed by #6074. Please submit a new issue if this continues to be a problem in 1.9.11+.
